pulsar-broker/src/test/java/org/apache/pulsar/broker/service/CurrentLedgerRolloverIfFullTest.java

```diff
@@ -104,12 +110,51 @@ public void testCurrentLedgerRolloverIfFull() throws Exception {
         stateUpdater.set(managedLedger, ManagedLedgerImpl.State.LedgerOpened);
         managedLedger.rollCurrentLedgerIfFull();
 
-        // the last ledger will be closed and removed and we have one ledger for empty
+        // If there are no pending write messages, the last ledger will be closed and still held.
         Awaitility.await()
                 .pollInterval(Duration.ofMillis(1000L))
                 .untilAsserted(() -> {
                     Assert.assertEquals(managedLedger.getLedgersInfoAsList().size(), 1);
-                    Assert.assertEquals(managedLedger.getTotalSize(), 0);
+                    Assert.assertNotEquals(managedLedger.getCurrentLedgerSize(), 0);
+                    MLDataFormats.ManagedLedgerInfo.LedgerInfo lastLhAfterRollover =
+                            managedLedger.getLedgersInfoAsList().get(managedLedger.getLedgersInfoAsList().size() - 1);
```

Comment:
Maybe we can add a `getLastLedgerId` method to simplify the code.

```java
        Supplier<Long> lastLedgerIdSupplier = () -> {
            final List<LedgerInfo> ledgers = managedLedger.getLedgersInfoAsList();
            Assert.assertFalse(ledgers.isEmpty());
            return ledgers.get(ledgers.size() - 1).getLedgerId();
        };
```

Reply:
Yes, I will change it ASAP.